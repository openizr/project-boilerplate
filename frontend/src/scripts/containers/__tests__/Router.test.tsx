/**
 * @jest-environment jsdom
 */

import * as React from 'react';
import Router from 'scripts/containers/Router';
import { act, render } from '@testing-library/react';

// Useful mocks allowing us to easily test React lazy components and Suspense.
jest.mock('react', () => {
  const MockedReact = jest.requireActual('react');
  MockedReact.Suspense = ({ children, fallback }: Any): Any => (
    process.env.LOADING === 'true' ? fallback : children
  );
  MockedReact.lazy = (callback: Any): Any => callback();
  return MockedReact;
});

jest.mock('scripts/store/routes', () => ({
  // eslint-disable-next-line global-require, @typescript-eslint/no-var-requires
  '/': (): Any => require('scripts/pages/Home').default,
}));

jest.mock('scripts/locale/index', () => jest.fn(() => Promise.resolve({ LABEL_TEST: 'Test' })));

jest.mock('scripts/store/index', () => ({
  mutate: jest.fn(),
  dispatch: jest.fn(),
  useCombiner: jest.fn((_hash, callback = (state: Any): Any => state) => callback({ route: (process.env.NOT_FOUND === 'true' ? null : '/') })),
}));

describe('containers/Router', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    delete process.env.LOADING;
    delete process.env.NOT_FOUND;
  });

  test('renders correctly - loading page', async () => {
    process.env.LOADING = 'true';
    let container: HTMLElement = document.createElement('div');
    await act(async () => {
      container = render(<Router />).container;
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - found page', async () => {
    let container: HTMLElement = document.createElement('div');
    await act(async () => {
      container = render(<Router />).container;
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - not found page', async () => {
    process.env.NOT_FOUND = 'true';
    let container: HTMLElement = document.createElement('div');
    await act(async () => {
      container = render(<Router />).container;
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
