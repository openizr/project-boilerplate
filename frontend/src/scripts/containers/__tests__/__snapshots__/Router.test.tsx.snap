// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/Router renders correctly - found page 1`] = `
<p>
  <img
    alt="test"
    src="file"
  />
  <img
    alt="test"
    src="file"
  />
  Test
</p>
`;

exports[`containers/Router renders correctly - loading page 1`] = `
<div>
  Loading...
</div>
`;

exports[`containers/Router renders correctly - not found page 1`] = `
<div>
  Not Found.
  object
</div>
`;
